spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ModuleStatusBuilder.java

```diff
@@ -56,9 +56,15 @@ ModuleStatusBuilder withId(ModuleDeploymentId id) {
 		return this;
 	}
 
-	// TODO: Find out what the application state candidates are
 	private static ModuleStatus.State convertState(String state) {
 		switch (state) {
+			case "STARTED":
+			case "STARTING":
+			case "STOPPED":
```

Comment:
I'd have to learn more about Lattice's approach, but it may not be worth the extra complexity it would expose. The biggest expense in a delete+deploy is the upload, and CF will cache that if the bits haven't changed (though we're not yet doing the upload in the right way to take advantage of that). We could certainly save several seconds a time by being able to `restart` rather than `redeploy` though, so worth looking into.